File: java-code/src/main/java/refactoring/refactoring_api/remove_argument_flag/before/Order.java
package refactoring.refactoring_api.remove_argument_flag.before;


import java.time.LocalDate;
import java.util.Arrays;
import java.util.List;

public class Order {

  String deliveryState;
  LocalDate placedOn;

    public static void main(String[] args) {
      var order = new Order();

      var rushDeliveryDate = order.deliveryDate( true);
      var regularDeliveryDate = order.deliveryDate( false);
    }

    public LocalDate deliveryDate(boolean isRush) {
      int deliveryTime;
      if (isRush) {
        if (Arrays.asList("MA", "CT").contains(this.deliveryState)) {
          deliveryTime = 1;
        } else if (Arrays.asList("NY", "NH").contains(this.deliveryState)) {
          deliveryTime = 2;
        } else {
          deliveryTime = 3;
        }
        return this.placedOn.plusDays(1 + deliveryTime);
      } else {
        if (Arrays.asList("MA", "CT", "NY").contains(this.deliveryState)) {
          deliveryTime = 2;
        } else if (Arrays.asList("ME", "NH").contains(this.deliveryState)) {
          deliveryTime = 3;
        } else {
          deliveryTime = 4;
        }
        return this.placedOn.plusDays(2 + deliveryTime);
      }
    }
  }

File: go-code/refactoring_api/remove_flag_argument/before/order.go
package before

import (
	"log"
	"time"
)

type Order struct {
	DeliveryState string
	PlacedOn      time.Time
}

func deliveryDate(anOrder Order, isRush bool) time.Time {
	var deliveryTime int
	if isRush {
		if contains([]string{"MA", "CT"}, anOrder.DeliveryState) {
			deliveryTime = 1
		} else if contains([]string{"NY", "NH"}, anOrder.DeliveryState) {
			deliveryTime = 2
		} else {
			deliveryTime = 3
		}
		return anOrder.PlacedOn.AddDate(0, 0, 1+deliveryTime)
	} else {
		if contains([]string{"MA", "CT", "NY"}, anOrder.DeliveryState) {
			deliveryTime = 2
		} else if contains([]string{"ME", "NH"}, anOrder.DeliveryState) {
			deliveryTime = 3
		} else {
			deliveryTime = 4
		}
		return anOrder.PlacedOn.AddDate(0, 0, 2+deliveryTime)
	}
}

func contains(slice []string, target string) bool {
	for _, item := range slice {
		if item == target {
			return true
		}
	}
	return false
}

func main() {
	order := Order{
		DeliveryState: "MA",
		PlacedOn:      time.Now(),
	}

	notRushDeliveryDate := deliveryDate(order, false)
	rushDeliveryDate := deliveryDate(order, true)

	log.Printf("notRushDeliveryDate = %v\n", notRushDeliveryDate)
	log.Printf("rushDeliveryDate = %v\n", rushDeliveryDate)
}
